File: src/main/java/mil/nga/tiff/internal/DirectoryStats.java
package mil.nga.tiff.internal;

import mil.nga.tiff.field.type.enumeration.DifferencingPredictor;
import mil.nga.tiff.field.type.enumeration.PlanarConfiguration;
import mil.nga.tiff.field.type.enumeration.SampleFormat;

import java.util.List;

public record DirectoryStats(
    Integer imageWidth,
    Integer imageHeight,
    Integer tileWidth,
    Integer tileHeight,
    int samplesPerPixel,
    List<Integer> bitsPerSample,
    List<SampleFormat> sampleFormatList,
    PlanarConfiguration planarConfiguration,
    List<Long> tileOffsets,
    List<Integer> tileByteCounts,
    List<Long> stripOffsets,
    List<Integer> stripByteCounts,
    Integer compression,
    DifferencingPredictor predictor
) {

}

File: src/main/java/mil/nga/tiff/internal/DirectoryStats.kt
package mil.nga.tiff.internal

import mil.nga.tiff.field.type.enumeration.Compression
import mil.nga.tiff.field.type.enumeration.DifferencingPredictor
import mil.nga.tiff.field.type.enumeration.PlanarConfiguration
import mil.nga.tiff.field.type.enumeration.SampleFormat

@JvmRecord
data class DirectoryStats(
    val imageWidth: Int?,
    val imageHeight: Int?,
    val tileWidth: Int?,
    val tileHeight: Int?,
    val samplesPerPixel: Int?,
    val bitsPerSample: List<Int>?,
    val sampleFormatList: List<SampleFormat>?,
    val planarConfiguration: PlanarConfiguration?,
    val tileOffsets: List<Long>?,
    val tileByteCounts: List<Int>?,
    val stripOffsets: List<Long>?,
    val stripByteCounts: List<Int>?,
    val compression: Compression,
    val predictor: DifferencingPredictor
)
